src/path_manager.c

```diff
@@ -1154,17 +1194,21 @@ struct mptcpd_pm *mptcpd_pm_create(struct mptcpd_config const *config)
                 return NULL;
         }
 
+        struct family_cb_data *const data = l_malloc(sizeof(*data));
+        data->config = config;
+        data->pm     = pm;
+
         if (l_genl_add_family_watch(pm->genl,
                                     name,
                                     family_appeared,
                                     family_vanished,
-                                    pm,
-                                    NULL) == 0
+                                    data,
+                                    l_free) == 0
             || !l_genl_request_family(pm->genl,
                                       name,
                                       family_appeared,
-                                      pm,
-                                      NULL)) {
+                                      data,
+                                      l_free)) {
```

Comment:
..._and_ here looks a little funny (in terms of how the ELL APIs work), but as far as I can see `l_genl_add_family_watch()` will not free `data` if it returns 0, and `l_genl_request_family()` will not free `data` if it returns `false`. But that also implies that `l_free(data)` is needed in the error handling block of code below.

Reply:
Good catch!  I'll add an `l_free()` to the error handling block.

Reply:
Looking at the code again, I am concerned about a potential double-`free()` on `data`.  I'll take a closer look.

Reply:
Commit 6b9de58 simplifies this patch set and addresses the memory management issues.  There is a longer a need to dynamically allocate and manage a convenience structure like  `family_cb_data` that introduced earlier.